app/controllers/hyrax/pages_controller.rb

```diff
@@ -46,5 +49,13 @@ def update_value_from_params
       def pages_layout
         action_name == 'show' ? 'homepage' : 'dashboard'
       end
+
+      # Use a registry of known ContentBlock names to whitelist which
+      # messages can be sent to the ContentBlock class
+      def page_from_params
+        name = params[:id]
+        return unless ContentBlock::NAME_REGISTRY.values.include?(name)
```

Comment:
Instead of using `public_send` directly, can we push this logic down into `ContentBlock` itself? `ContentBlock.instance(name)` or something?